File: ios/CodeScanner/ResultManager.m
//
//  ResultManager.m
//  RNSCanner
//
//  Created by jacokao on 2018/7/25.
//  Copyright © 2018年 Facebook. All rights reserved.
//


#import <React/RCTBridgeModule.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(ResultManager, NSObject)

RCT_EXTERN_METHOD(checkValue:(NSNumber) dur
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject
                  )

@end

File: ios/CodeScanner/ResultManager.swift
//
//  ResultManager.swift
//  RNSCanner
//
//  Created by jeff on 2018/7/25.
//  Copyright © 2018年 Facebook. All rights reserved.
//

import Foundation

@objc(ResultManager)
class ResultManager: RCTEventEmitter {
  // ...
  @objc func checkValue(_ dur: NSNumber,
                        resolver resolve: RCTPromiseResolveBlock,
                        rejecter reject: RCTPromiseRejectBlock) -> Void {
    
    print("check value")
    var mode = -1
    while(mode != 2) {
      mode = UserDefaults.standard.integer(forKey: "RunStep")
      print(mode)
      if(mode == 2) {
        break
        
      }
      usleep(200000)
      print(mode)
    }
    print("scan value ")
    let scanValue = UserDefaults.standard.string(forKey: "ScannedValue")
    print(scanValue)
    resolve(scanValue)
  }
  // ...
}
